controllers/external-accounts.js

```diff
@@ -90,10 +91,26 @@ const syncExternalAccountData = async (req, res) => {
 
     await Promise.all(updateUserDataPromises);
 
+    const inDiscordUsers = await getUsersByRole("in_discord");
+    inDiscordUsers.forEach((user) => {
+      if (user.roles.archived === true) {
+        const userData = {
+          roles: {
+            ...user.roles,
+            archived: false,
+          },
+        };
+        updateArchivedPromises.push(addOrUpdate(userData, user.id));
+      }
+    });
+
+    await Promise.all(updateArchivedPromises);
```

Comment:
Do we need two `Prosime.all`? One here and other in line 92

Reply:
Yes as its an array of promises and if we wait inside loop for each user then the api wait time will increase drastically, 
promise.all resolves all promises without waiting for previpus one to finish, i.e. dosent matter which promise resolves first as soon as all the promises are resolved the await on promise.all will move ahead in the thread.